app/src/main/java/mobi/inthepocket/android/beacons/app/service/BeaconActivityService.java

```diff
@@ -0,0 +1,79 @@
+package mobi.inthepocket.android.beacons.app.service;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.v4.app.JobIntentService;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.NotificationManagerCompat;
+import android.util.Log;
+import android.widget.Toast;
+
+import mobi.inthepocket.android.beacons.app.R;
+import mobi.inthepocket.android.beacons.ibeaconscanner.Beacon;
+import mobi.inthepocket.android.beacons.ibeaconscanner.BluetoothScanBroadcastReceiver;
+
+/**
+ * Target service responsible for receiving beacon updates from the library (and communicating them).
+ */
+public class BeaconActivityService extends JobIntentService
```

Comment:
Are we sure that this also works with a regular service? or will we have to change the Harald service to a JobIntentService?

Reply:
As far as my understanding goes: Normal services can only run in the foreground / will have a much shorter window when being executed in the background.

JobIntentService is now the way to go do short-lived work.

Reply:
What the man said.
I don't see what made you think this JobIntentService implementation also works with a regular Service, @qbraet ?

In my testing, now a bit over four weeks ago, using a Service had a few drawbacks (such as a persistent notification for long-running operations) and was unable to achieve the desired result. JobIntentService is now the approach promoted by Google for short-lived work, indeed.